packages/editor-ui/src/stores/evaluations.store.ee.ts

```diff
@@ -0,0 +1,184 @@
+import { defineStore } from 'pinia';
+import { computed, ref } from 'vue';
+import { useRootStore } from './root.store';
+import * as testDefinitionsApi from '@/api/evaluations.ee';
+import type { ITestDefinition } from '@/api/evaluations.ee';
+import { usePostHog } from './posthog.store';
+import { STORES, WORKFLOW_EVALUATION_EXPERIMENT } from '@/constants';
+
+export const useEvaluationsStore = defineStore(
+	STORES.EVALUATIONS,
+	() => {
+		// State
+		const testDefinitionsById = ref<Record<number, Partial<ITestDefinition>>>({});
+		const loading = ref(false);
+		const fetchedAll = ref(false);
+
+		// Store instances
+		const posthogStore = usePostHog();
+		const rootStore = useRootStore();
+
+		// Computed
+		const allTestDefinitions = computed(() => {
+			return Object.values(testDefinitionsById.value).sort((a, b) =>
+				(a.name ?? '').localeCompare(b.name ?? ''),
+			);
+		});
+
+		// Enable with `window.featureFlags.override('025_workflow_evaluation', true)`
+		const isFeatureEnabled = computed(() =>
+			posthogStore.isFeatureEnabled(WORKFLOW_EVALUATION_EXPERIMENT),
+		);
+
+		const isLoading = computed(() => loading.value);
+
+		const hasTestDefinitions = computed(() => Object.keys(testDefinitionsById.value).length > 0);
+
+		// Methods
+		const setAllTestDefinitions = (definitions: ITestDefinition[]) => {
+			testDefinitionsById.value = definitions.reduce(
+				(acc: Record<number, ITestDefinition>, def: ITestDefinition) => {
+					acc[def.id] = def;
+					return acc;
+				},
+				{},
+			);
+			fetchedAll.value = true;
+		};
+
+		/**
+		 * Upserts test definitions in the store.
+		 * @param toUpsertDefinitions - An array of test definitions to upsert.
+		 */
+		const upsertTestDefinitions = (toUpsertDefinitions: Array<Partial<ITestDefinition>>) => {
+			toUpsertDefinitions.forEach((toUpsertDef) => {
+				const defId = toUpsertDef.id;
+				if (!defId) throw Error('ID is required for upserting');
+				const currentDef = testDefinitionsById.value[defId];
+				if (currentDef) {
+					testDefinitionsById.value = {
+						...testDefinitionsById.value,
+						[defId]: {
+							...currentDef,
+							...toUpsertDef,
+						},
+					};
+				} else {
+					testDefinitionsById.value = {
+						...testDefinitionsById.value,
+						[defId]: toUpsertDef,
+					};
+				}
```

Comment:
It's a matter of preference, but undefined in spread syntax doesn't do anything, so the code may be simplified to
```suggestion
					testDefinitionsById.value = {
						...testDefinitionsById.value,
						[defId]: {
							...currentDef,
							...toUpsertDef,
						},
					};
```

Reply:
Good point! Thanks!